extensions/amp-story/1.0/test/test-live-story-manager.js

```diff
@@ -0,0 +1,108 @@
+  let ampStory;
+  let storyEl;
+
+  /**
+   * @param {!Element} container
+   * @param {number} count
+   * @param {Array<string>=} opt_ids
+   * @return {!Array<!Element>}
+   */
+  function createPages(container, count, opt_ids) {
+    return Array(count).fill(undefined).map((unused, i) => {
+      const page = win.document.createElement('amp-story-page');
+      page.id = opt_ids && opt_ids[i] ? opt_ids[i] : `-page-${i}`;
+      const storyPage = new AmpStoryPage(page);
+      page.getImpl = () => Promise.resolve(storyPage);
+      sandbox.stub(storyPage, 'mutateElement').callsFake(fn => fn());
+      container.appendChild(page);
+      return page;
+    });
+  }
+
+  beforeEach(() => {
+    win = env.win;
+    storyEl = win.document.createElement('amp-story');
+    addAttributesToElement(storyEl,
+        {'id': 'testStory', 'dynamic-live-list': 'storyLiveList'});
+    ampStory = new AmpStory(storyEl);
+    liveStoryManager = new LiveStoryManager(ampStory);
+  });
+
+  it('should build a dynamic live-list', () => {
+    liveStoryManager.build();
+
+    const liveListEl = ampStory.element.querySelector('amp-live-list');
+    expect(liveListEl).to.exist;
+  });
+
+  it('live-list id should correspond to attribute dynamic-live-list', () => {
+    liveStoryManager.build();
+    const storyAttr = ampStory.element.getAttribute('dynamic-live-list');
+
+    const liveListEl = ampStory.element.querySelector('amp-live-list');
+    expect(storyAttr).to.equal(liveListEl.id);
+  });
+
+  it('should throw if no dynamic-live-list attr is set', () => {
+    ampStory.element.setAttribute('dynamic-live-list', '');
+
+    allowConsoleError(() => { expect(() => {
+      liveStoryManager.build();
+    }).to.throw(/Story must contain dynamic-live-list to build an amp-live-list/); });
+  });
+
+  it('should throw if no story id is set', () => {
+    ampStory.element.setAttribute('id', '');
+
+    allowConsoleError(() => { expect(() => {
+      liveStoryManager.build();
+    }).to.throw(/Story must contain id to build an amp-live-list/); });
+  });
+
+
+  it('should append new page from server on update', () => {
+    const currentPages = createPages(ampStory.element, 2, ['cover', 'page-1']);
+    expect(ampStory.element.children.length).to.equal(2);
+
+    const fromServerStoryEl = win.document.createElement('amp-story');
+    const fromServerStory = new AmpStory(fromServerStoryEl);
+    const fromServerPages =
+      createPages(fromServerStory.element, 2, ['cover', 'page-1', 'page-3']);
+    const fromServerLastPage = fromServerPages[fromServerPages.length - 1];
+    fromServerLastPage.classList.add('amp-live-list-item-new');
+
+    liveStoryManager.build();
+    liveStoryManager.update(fromServerStory.element, currentPages);
+
+    expect(ampStory.element.children.length).to.equal(3);
+
+    const fromClientLastPage =
+        ampStory.element.querySelector('amp-story-page:last-of-type');
+
+    expect(fromClientLastPage.id).to.equal(fromServerLastPage.id);
```

Comment:
Can you please split this in two tests: one to check that a page got appended, and the other to check its position?